readthedocs/templates/search/elastic_search.html

```diff
@@ -179,61 +179,64 @@ <h3>
                     {% endfor %}
 
                   {% elif 'page' in result.meta.index %}
-                    <a href="{% doc_url result.project|get_project result.version result.full_path %}?highlight={{ query }}">
-                      {{ result.project }} - {% if result.meta.highlight.title %} {{ result.meta.highlight.title.0|safe }} {% else %} {{ result.title }} {% endif %}
-                    </a>
-
-                    {% for inner_hit in result.meta.inner_hits %}
-                      {% if inner_hit.type == 'domains' %}
-                        <p>
-                          <a href="{% doc_url result.project|get_project result.version inner_hit.source.doc_name %}?highlight= {{ query }}#{{ inner_hit.source.anchor }}">
-
-                            {% if inner_hit.source.display_name|length >= 1 %}
-                              ({{ inner_hit.source.role_name }}) {{ inner_hit.source.display_name}}
+                    {% with "100" as MAX_SUBSTRING_LIMIT %}
```

Comment:
I'm feeling the pain of maintaining multiple sets of result HTML. We should consider trying to get the site search using the search as you type code, hopefully with minimal changes from the theme code. That way we don't have 3 places to change HTML. 

The goal will be to ship search as you type across all of RTD so we only have to maintain one set of results, but baby steps :)

Reply:
Yes... maintaining multiple sets of html is definitely a pain.

> The goal will be to ship search as you type across all of RTD so we only have to maintain one set of results, but baby steps :)

I think that's quite a big refactor and we can target that after this PR?